src/concrete/ml/sklearn/base.py

```diff
@@ -2080,6 +2079,9 @@ def post_processing(self, y_preds: numpy.ndarray) -> numpy.ndarray:
         Returns:
             numpy.ndarray: The majority vote.
         """
+
+        self.topk = y_preds.squeeze()
```

Comment:
this is weird no ? I don't see the KNN having this attribute anywhere else than here !

Reply:
I saved it to get the topk-labels returned by _inference and compare them with those of scikit learn.
The reason I squeeze it is that _inference returns a tuple, so it will return a tensor of (X_test, 1, n_neighorbs).
I squeezed it to get a tensor of (X_test, n_neighorbs).

Reply:
what I meant is that attributes should not be exposed like that without being defined in the init ! But more than that, why is this needed ? is topk something exposed by the scikit-learn model ?  if so, only after running the predict method ? If not, then I would not add this attribute in CML at all

Reply:
Yes, it's exposed by scikit-learn, through:  `distance, topk_args = sklearn_model.kneighbors(X_test)` after performing the prediction.
The reason that i added it, is to compare it with the one of SK as discussed in weekly.
I'll add it to the init.

Reply:
I'm not sure I get it , can we do `sklearn_model.topk` ? and does it work before calling `predict` ? if you are saying that it's available by calling the `kneighbors` method then it would be prefereable to have such a method. And if it requires to store predictions /topk in the class somewhere (which I still find intriguing) then it's probably done through a private attribute (which is better at least) ! 

Also, be careful of double fitting, serialization and deployment tests, I think it should be fine as it's made in post-processing but still !

Reply:
I'll close this comment, as we agreed to set a new method called get_topk_labels.